### PR TITLE
[SDKS-7496] remove all flags when checkFilterQuery

### PR DESCRIPTION
**Pull request by emmaz90**
# Javascript commons library

## Extra Notes

### CODE REVIEW

File: src/storages/inLocalStorage/SplitsCacheInLocal.ts [hunk 1]
```diff
@@ -251,19 +245,8 @@ export class SplitsCacheInLocal extends AbstractSplitsCacheSync {
...
-              groupedFilters.byPrefix.some((prefix: string) => splitName.startsWith(prefix + '__'))
-            )) {
-              // * set `cacheReadyButNeedsToFlush` so that `checkCache` returns true (the storage is ready to be used) and the data is cleared before updating on first successful splits fetch
-              this.cacheReadyButNeedsToFlush = true;
-              return;
-            }
-            this.removeSplit(splitName);
-          });
+          // * Remove all splits from cache
+          this.removeSplits(this.getSplitNames());
```
Comment (EmilianoSanchez) [inline src/storages/inLocalStorage/SplitsCacheInLocal.ts:249]: Check this line. I think this should be `this.clear();` to clear all the cache. You will need to update the E2E tests to see if the behaviour is not broken and it passes edge cases.

**Review by EmilianoSanchez — CHANGES_REQUESTED** [review]

File: src/storages/inLocalStorage/SplitsCacheInLocal.ts [hunk 2]
```diff
@@ -251,19 +245,8 @@ export class SplitsCacheInLocal extends AbstractSplitsCacheSync {
...
-              groupedFilters.byName.indexOf(splitName) > -1 ||
-              groupedFilters.byPrefix.some((prefix: string) => splitName.startsWith(prefix + '__'))
-            )) {
-              // * set `cacheReadyButNeedsToFlush` so that `checkCache` returns true (the storage is ready to be used) and the data is cleared before updating on first successful splits fetch
-              this.cacheReadyButNeedsToFlush = true;
-              return;
-            }
-            this.removeSplit(splitName);
-          });
+          // * clear cache
```
Comment (EmilianoSanchez) [inline src/storages/inLocalStorage/SplitsCacheInLocal.ts:248]: // if there is cache, clear it
        if (this.checkCache()) this.clear();

File: src/storages/inLocalStorage/SplitsCacheInLocal.ts [hunk 3]
```diff
@@ -251,19 +245,8 @@ export class SplitsCacheInLocal extends AbstractSplitsCacheSync {
           // * set change number to -1, to fetch splits with -1 `since` value.
           localStorage.setItem(this.keys.buildSplitsTillKey(), '-1');
```
Comment (EmilianoSanchez) [inline src/storages/inLocalStorage/SplitsCacheInLocal.ts:246]: We can remove this line, since the `clear` call will remove the item anyway.

**Review by EmilianoSanchez — APPROVED** [review]
